lib/Conversion/FIRRTLToHW/LowerToHW.cpp

```diff
@@ -2454,207 +2464,35 @@ LogicalResult FIRRTLLowering::visitDecl(NodeOp op) {
-    };
-    recurse(reg, type, "");
-  };
-
-  // Emit the initializer expression for simulation that fills it with random
-  // value.
-  addToIfDefBlock("SYNTHESIS", std::function<void()>(), [&]() {
-    addToIfDefBlock(
-        "RANDOMIZE_REG_INIT",
-        [&]() { regInsertionPoint = builder.saveInsertionPoint(); },
-        std::function<void()>());
-    addToInitialBlock([&]() {
-      emitRandomizePrologIfNeeded();
-      circuitState.used_RANDOMIZE_REG_INIT = 1;
-      auto *block = builder.getBlock();
-
-      // Randomized values are assigned to registers in `ifdef
-      // RANDOMIZE_REG_INIT block.
-      auto &op = randomizeRegInitIfOp[block];
-      if (!op)
-        op = builder.create<sv::IfDefProceduralOp>("RANDOMIZE_REG_INIT",
-                                                   [&]() {});
-      runWithInsertionPointAtEndOfBlock(randomInit, op.thenRegion());
-
-      // If the register is async reset, we need to insert extra initialization
-      // in post-randomization so that we can set the reset value to register if
-      // the reset signal is enabled.
-      if (asyncRegResetInitPair.hasValue()) {
-        Value resetSignal, resetValue;
-        std::tie(resetSignal, resetValue) = *asyncRegResetInitPair;
-        // Merge if op if their reset values are same.
-        auto &op = asyncRegPostRandomizationIfOp[{block, resetSignal}];
-        if (!op)
-          builder.create<sv::IfDefProceduralOp>("RANDOMIZE", [&] {
-            op = builder.create<sv::IfOp>(resetSignal, [&]() {});
-          });
-
-        runWithInsertionPointAtEndOfBlock(
-            [&]() { builder.create<sv::BPAssignOp>(reg, resetValue); },
-            op.thenRegion());
-      }
-    });
-  });
-}
-
 LogicalResult FIRRTLLowering::visitDecl(RegOp op) {
   auto resultType = lowerType(op.result().getType());
   if (!resultType)
     return failure();
   if (resultType.isInteger(0))
     return setLowering(op, Value());
 
+  Value clockVal = getLoweredValue(op.clockVal());
+  if (!clockVal)
+    return failure();
+
   // Add symbol if DontTouch annotation present.
-  auto symName = op.inner_symAttr();
+  auto innerSymAttr = op.inner_symAttr();
   if (AnnotationSet::removeAnnotations(
           op, "firrtl.transforms.DontTouchAnnotation") &&
-      !symName)
-    symName = op.nameAttr();
-  auto regResult =
-      builder.create<sv::RegOp>(resultType, op.nameAttr(), symName);
-  (void)setLowering(op, regResult);
-
-  initializeRegister(regResult);
-
+      !innerSymAttr)
+    innerSymAttr = op.nameAttr();
+  if (!innerSymAttr)
+    innerSymAttr = builder.getStringAttr(
+        moduleNamespace.newName(Twine("__") + op.name() + "__"));
+
+  // Create a reg op, wiring itself to its input.
+  Backedge inputEdge = backedgeBuilder.get(resultType);
+  auto reg = builder.create<seq::FirRegOp>(inputEdge, clockVal, op.nameAttr(),
+                                           innerSymAttr, /*isRandom=*/true);
```

Comment:
nit: i don't know if tools look at these little helper comments, but I think this should say `isRandomized` to be fully explicit.

Reply:
Thanks for spotting this.